clients/linux/src/account.rs

```diff
@@ -361,11 +403,107 @@ impl Editor {
         textarea.set_left_margin(4);
         textarea.set_tab_width(4);
 
+        let target_list = TargetList::new(&[]);
+
+        textarea.drag_dest_set(DestDefaults::ALL, &[], DragAction::COPY);
+
+        target_list.add_uri_targets(URI_TARGET_INFO);
+        target_list.add_text_targets(TEXT_TARGET_INFO);
+        target_list.add_image_targets(IMAGE_TARGET_INFO, true);
+
+        textarea.drag_dest_set_target_list(Some(&target_list));
+
+        textarea.connect_drag_data_received(Self::on_drag_data_received(m));
+        textarea.connect_paste_clipboard(Self::on_paste_clipboard(m));
+        textarea.connect_button_press_event(Self::on_button_press(m));
+
         let textview = textarea.upcast_ref::<gtk::TextView>();
+
         let gspell_view = gspell::TextView::get_from_gtk_text_view(textview).unwrap();
         gspell_view.basic_setup();
 
-        textarea.connect_button_press_event(closure!(m => move |w, evt| {
+        let scroll = GtkScrolledWindow::new(None::<&GtkAdjustment>, None::<&GtkAdjustment>);
+        scroll.add(&textarea);
+
+        let cntr = GtkStack::new();
+        cntr.add_named(&empty, "empty");
+        cntr.add_named(&info, "folderinfo");
+        cntr.add_named(&scroll, "scroll");
+
+        let highlighter = LanguageManager::get_default().unwrap_or_default();
+        let lang_paths = highlighter.get_search_path();
+
+        let mut str_paths: Vec<&str> = lang_paths.iter().map(|path| path.as_str()).collect();
+        let lang_specs = get_language_specs_dir();
+        str_paths.push(lang_specs.as_str());
+        highlighter.set_search_path(str_paths.as_slice());
+
+        Self {
+            info,
+            textarea,
+            highlighter,
+            change_sig_id: RefCell::new(None),
+            cntr,
+            messenger: m.clone(),
+        }
+    }
+
+    fn on_drag_data_received(
+        m: &Messenger,
+    ) -> impl Fn(&GtkSourceView, &DragContext, i32, i32, &SelectionData, u32, u32) {
+        closure!(m => move |_, _, _, _, s, info, _| {
+            let target = match info {
+                URI_TARGET_INFO => {
+                    TextAreaDropPasteInfo::Uris(s.get_uris().iter().map(|uri| uri.to_string()).collect())
+                }
+                IMAGE_TARGET_INFO => match s.get_pixbuf() {
+                    None => {
+                        m.send_err_dialog("Dragging bytes", uerr_dialog!("Unsupported format"));
+                        return;
+                    }
+                    Some(pixbuf) => match pixbuf.save_to_bufferv("jpg", &[]) {
+                        Ok(bytes) => TextAreaDropPasteInfo::Image(bytes),
+                        Err(err) => {
+                            m.send_err_dialog("Dragging bytes", LbError::fmt_program_err(err));
+                            return;
+                        }
+                    },
+                },
+                TEXT_TARGET_INFO => return,
+                _ => panic!("impossible"),
```

Comment:
idk if this is panic worthy, you could just `eprintln`, pop a dialog open, or return early from the closure